tests/TestCase/TestSuite/Fixture/SchemaLoaderTest.php

```diff
@@ -0,0 +1,135 @@
+    {
+        parent::setUp();
+        $this->restore = $GLOBALS['__PHPUNIT_BOOTSTRAP'];
+        unset($GLOBALS['__PHPUNIT_BOOTSTRAP']);
+
+        $this->loader = new SchemaLoader(['outputLevel' => ConsoleIo::QUIET]);
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+        $GLOBALS['__PHPUNIT_BOOTSTRAP'] = $this->restore;
+
+        (new SchemaCleaner())->dropTables('test', ['schema_loader_test_one', 'schema_loader_test_two']);
+        ConnectionManager::drop('schema_test');
+    }
+
+    /**
+     * Tests loading schema files.
+     */
+    public function testLoadingFiles(): void
+    {
+        $connection = ConnectionManager::get('test');
+
+        $schemaFiles[] = $this->createSchemaFile('schema_loader_test_one');
+        $schemaFiles[] = $this->createSchemaFile('schema_loader_test_two');
+
+        $this->loader->loadFiles($schemaFiles, 'test', false, false);
+
+        $connection = ConnectionManager::get('test');
+        $tables = $connection->getSchemaCollection()->listTables();
+        $this->assertContains('schema_loader_test_one', $tables);
+        $this->assertContains('schema_loader_test_two', $tables);
+    }
+
+    /**
+     * Tests loading missing files.
+     */
+    public function testLoadMissingFile(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->loader->loadFiles('missing_schema_file.sql', 'test', false, false);
+    }
+
+    /**
+     * Tests dropping and truncating tables during schema load.
+     */
+    public function testDropTruncateTables(): void
+    {
+        $connection = $this->createMock(Connection::class);
+
+        $tableSchema = $this->createMock(TableSchema::class);
+        $schemaDialect = $this->createMock(SchemaDialect::class);
+        $schemaDialect
+            ->expects($this->atLeastOnce())->method('dropConstraintSql')->with($tableSchema)->willReturn(['']);
+        $schemaDialect
+            ->expects($this->atLeastOnce())->method('dropTableSql')->with($tableSchema)->willReturn(['']);
+        $schemaDialect
+            ->expects($this->atLeastOnce())->method('truncateTableSql')->with($tableSchema)->willReturn(['']);
+        $driver = $this->createMock(DriverInterface::class);
+        $driver
+            ->expects($this->atLeastOnce())->method('schemaDialect')->willReturn($schemaDialect);
+        $connection
+            ->expects($this->atLeastOnce())->method('getDriver')->willReturn($driver);
+
+        $schemaCollection = $this->createMock(Collection::class);
+        $schemaCollection
+            ->expects($this->atLeastOnce())->method('listTables')->willReturn(['schema_test']);
+        $schemaCollection
+            ->expects($this->atLeastOnce())->method('describe')->willReturn($tableSchema);
+        $connection
+            ->expects($this->atLeastOnce())->method('getSchemaCollection')->willReturn($schemaCollection);
+
+        ConnectionManager::setConfig('schema_test', $connection);
```

Comment:
Could we create a table with a random name, and drop/truncate that instead? This many mocks makes the test very brittle and tedious to maintain as the test is coupled to the implementation tightly.